lib/compiler-singlepass/src/machine.rs

```diff
@@ -35,68 +46,110 @@ impl Machine {
         self.stack_offset.0
     }
 
+    fn get_used_in(mut v: u32) -> Vec<u8> {
```

Comment:
This could grab a mapping closure and return `Vec<RegisterEnum>` directly, to avoid an intermediate vec allocation.

Although this probably reuses the allocation due to the specializations that exist, it seems pretty brittle (e.g. `repr(u8)` might get changed or somebody could fail to understand `u8` here has a purpose.

Reply:
Good point! I had the same thought about specializations, but didn’t think of just making `get_used_in` take a closure :)